crates/biome_js_semantic/src/events.rs

```diff
@@ -352,41 +359,59 @@ impl SemanticEventExtractor {
                 }
                 binding.ident_token()
             }
-        };
-        let name_token = name_token.ok()?;
+        }
+        .ok()?;
 
         let node = node.syntax();
         let parent = node.parent()?;
         let parent_kind = parent.kind();
         match parent_kind {
             JS_VARIABLE_DECLARATOR => {
-                if let Some(true) = Self::is_var(&parent) {
-                    let hoisted_scope_id = self.scope_index_to_hoist_declarations(0);
-                    self.push_binding_into_scope(hoisted_scope_id, &name_token, parent_kind);
+                let declarator = JsVariableDeclarator::unwrap_cast(parent);
+                let declaration = declarator.declaration()?;
+                let hoisted_scope_id = if declaration.is_var() {
+                    self.scope_index_to_hoist_declarations(0)
                 } else {
-                    self.push_binding_into_scope(None, &name_token, parent_kind);
+                    None
                 };
-                self.export_variable_declarator(node, &parent);
+                self.push_binding_into_scope(
+                    hoisted_scope_id,
+                    &name_token,
+                    parent_kind,
+                    DeclarationKind::Value,
+                );
+                self.export_variable_declaration(node, &declaration);
             }
             JS_FUNCTION_DECLARATION | JS_FUNCTION_EXPORT_DEFAULT_DECLARATION => {
                 let hoisted_scope_id = self.scope_index_to_hoist_declarations(1);
-                self.push_binding_into_scope(hoisted_scope_id, &name_token, parent_kind);
+                self.push_binding_into_scope(
```

Comment:
nit: I think I would like to have three functions now:
`push_value_binding_into_scope`, `push_type_binding_into_scope` and `push_dual_binding_into_scope`. Less prone to error because we don't need to pass `DeclarationKind`.